tests/Unit/NumericalAlgorithms/SphericalHarmonics/Test_TensorYlmCartToSphere.cpp

```diff
@@ -0,0 +1,469 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <filesystem>
+#include <fstream>
+#include <optional>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/SimpleSparseMatrix.hpp"
+#include "DataStructures/Tensor/Structure.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "NumericalAlgorithms/SphericalHarmonics/TensorYlmCartToSphere.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace {
+
+template <typename TensorStructure>
+std::string symm_to_string() {
+  if constexpr (std::is_same_v<typename TensorStructure::symmetry,
+                               Symmetry<1>>) {
+    return "a";
+  } else if constexpr (std::is_same_v<typename TensorStructure::symmetry,
+                                      Symmetry<1, 1>>) {
+    return "aa";
+  } else if constexpr (std::is_same_v<typename TensorStructure::symmetry,
+                                      Symmetry<2, 1>>) {
+    return "ab";
+  } else if constexpr (std::is_same_v<typename TensorStructure::symmetry,
+                                      Symmetry<3, 2, 1>>) {
+    return "abc";
+  } else if constexpr (std::is_same_v<typename TensorStructure::symmetry,
+                                      Symmetry<2, 1, 1>>) {
+    return "abb";
+  }
+  return "";
+}
+
+template <typename TensorStructure, typename SparseMatrixType>
+void test_tensorylm_cart_to_sphere_vs_spec(const size_t ell_max) {
+  // There are two "modes" for this test.  If test_all_elements is
+  // true, it tests all the matrix elements vs SpEC, reading .txt
+  // files that were output by SpEC. This test was done by Mark
```

Comment:
1. Is the code to write the txt file commited to spec master? If not, should we do that?
2. Do you have an estimate of the file size and runtime of the test, just so folks in the future can understand why it's unreasonable to do this in CI?